packages/@aws-cdk/aws-dynamodb/lib/table-time-to-live-handler/index.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable no-console */
+import type { IsCompleteRequest, IsCompleteResponse, OnEventRequest, OnEventResponse } from '@aws-cdk/custom-resources/lib/provider-framework/types';
+import { DynamoDB } from 'aws-sdk'; // eslint-disable-line import/no-extraneous-dependencies
+import { disableTimeToLive, enableTimeToLive, isTimeToLiveCorrect, isTimeToLiveStable } from './utils';
+
+const dynamodb = new DynamoDB({ apiVersion: '2012-08-10' });
+
+export async function onEventHandler(event: OnEventRequest): Promise<OnEventResponse> {
+  console.log('Event: %j', event);
+
+  if (event.RequestType === 'Create' || event.RequestType === 'Update') {
+    const currentTtl = await dynamodb.describeTimeToLive({
+      TableName: event.ResourceProperties.TableName,
+    }).promise();
+    console.log('Describe time to live: %j', currentTtl);
+
+    switch (currentTtl.TimeToLiveDescription?.TimeToLiveStatus) {
+      case 'ENABLING':
+        throw new Error('Can not change time to live while it is being enabled.');
+        break;
+      case 'DISABLING':
+        throw new Error('Can not change time to live while it is being disabled.');
+        break;
+      case 'ENABLED':
+        if (event.ResourceProperties.TimeToLiveAttribute !== currentTtl.TimeToLiveDescription.AttributeName) {
+          await disableTimeToLive(event);
+        }
+        break;
+      case 'DISABLED':
+        if (event.ResourceProperties.TimeToLiveAttribute) {
+          await enableTimeToLive(event);
+        }
+        break;
+    }
+
+    console.log('Update table: %j', event.ResourceProperties.TableName);
+    return { PhysicalResourceId: event.PhysicalResourceId ?? event.LogicalResourceId };
+  }
+
+  return { PhysicalResourceId: event.PhysicalResourceId };
+}
+
+export async function isCompleteHandler(event: IsCompleteRequest): Promise<IsCompleteResponse> {
+  console.log('Event: %j', event);
+
+  const data = await dynamodb.describeTable({
+    TableName: event.ResourceProperties.TableName,
+  }).promise();
+  console.log('Describe table: %j', data);
+
+  const tableActive = (data.Table?.TableStatus === 'ACTIVE');
+
+  if (await isTimeToLiveStable(event)) {
+    await onEventHandler(event);
+  }
```

Comment:
I don't get why you're calling the other handler here...

Reply:
I do this, so we can allow the scenario of having a ttl set and then “directly” changing it to a new value. For this to work the ttl needs to be disabled first. Afterwards I want to enable the ttl with the new value. The event handler knows, that if the ttl values are different the ttl needs to be disabled in all scenarios. If this has happened `isTimeToLiveStable` will return true again. Then we simply call `onEventHandler` again and the logic for going from disabled to enabled will do step two. So, this is basically to reduce code duplication and reuse the existing logic.

If you’d like I can add a comment to this line describing, why we call it again.

Reply:
Got it! Yes I think this is worth a comment or maybe repeat some code to make it clear.

Shouldn't this be moved inside the `switch` statement? You don't need to call this for a `Delete` event.

Also by repeating some code/logic there you can simply `return { isComplete: false }` when you know that you still need to enabled/disable the TTL instead of introducing `isTimeToLiveCorrect()` to avoid a race condition? Will reduce the number of API calls when you don't need to enable/disable.

Reply:
Agreed that this absolutely deserves a comment in the code 🙂.

Reply:
I have added a comment. @skinny85 please let me know if this is clear enough or if it needs to be reworded.